### ISSUE TITLE
Some modifications needed

### ISSUE THREAD

**Issue by taewookim** [yeephycho/widerface-to-tfrecord#1]
There was some errors.
1) location of annotation

> f = open("WIDER/wider_face_train_annot.txt")

I noticed in the Face annotations came in different file.

This solved it:

> f = open("WIDER/wider_face_split/wider_face_train_bbx_gt.txt")

2) Encoding issues

Some lines needed UTF-8 encoding in tf.train.Example() line

Original:

>   tf_example = tf.train.Example(features=tf.train.Features(feature={
>     'image/height': dataset_util.int64_feature(int(height)),
>     'image/width': dataset_util.int64_feature(int(width)),
>     'image/filename': dataset_util.bytes_feature(filename),
>     'image/source_id': dataset_util.bytes_feature(filename),
>     'image/key/sha256': dataset_util.bytes_feature(key.encode('utf8')),
>     'image/encoded': dataset_util.bytes_feature(encoded_image_data),
>     'image/format': dataset_util.bytes_feature('jpeg'.encode('utf8')),
>     'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
>     'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
>     'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
>     'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
>     'image/object/class/text': dataset_util.bytes_list_feature(classes_text),
>     'image/object/class/label': dataset_util.int64_list_feature(classes),
>     'image/object/difficult': dataset_util.int64_list_feature(int(0)),
>     'image/object/truncated': dataset_util.int64_list_feature(truncated),
>     'image/object/view': dataset_util.bytes_list_feature(poses),
>     }))

Updated code

 

>  tf_example = tf.train.Example(features=tf.train.Features(feature={
>     'image/height': dataset_util.int64_feature(int(height)),
>     'image/width': dataset_util.int64_feature(int(width)),
>     'image/filename': dataset_util.bytes_feature(filename.encode('utf8')),
>     'image/source_id': dataset_util.bytes_feature(filename.encode('utf8')),
>     'image/key/sha256': dataset_util.bytes_feature(key.encode('utf8')),
>     'image/encoded': dataset_util.bytes_feature(encoded_image_data),
>     'image/format': dataset_util.bytes_feature('jpeg'.encode('utf8')),
>     'image/object/bbox/xmin': dataset_util.float_list_feature(xmins),
>     'image/object/bbox/xmax': dataset_util.float_list_feature(xmaxs),
>     'image/object/bbox/ymin': dataset_util.float_list_feature(ymins),
>     'image/object/bbox/ymax': dataset_util.float_list_feature(ymaxs),
>     'image/object/class/text': dataset_util.bytes_list_feature([x.encode('utf8') for x in classes_text]),
>     'image/object/class/label': dataset_util.int64_list_feature(classes),
>     'image/object/difficult': dataset_util.int64_list_feature(int(0)),
>     'image/object/truncated': dataset_util.int64_list_feature(truncated),
>     'image/object/view': dataset_util.bytes_list_feature(poses),
>     }))

**Comment by yeephycho** [comment]
Thank you very much for your comment!
Gonna modify the encoding part.
